src/lexer/lexer.rs

```diff
@@ -1011,9 +1036,17 @@ impl<'a, PC: PreprocContext> Lexer<'a, PC> {
         }
     }
 
+    pub(super) fn span(&self) -> Span {
+        Span {
+            file: self.buf.get_source_id(),
+            start: self.start.unwrap(),
+            end: self.location(),
+        }
+    }
+
     pub fn next_token(&mut self) -> LocToken {
         loop {
-            let start = self.location();
+            self.start = Some(self.location());
```

Comment:
Why are you changing this ?
And I don't see the point to have this value in an option to unwrap it after.

Reply:
I'm storing start in the lexer so that I don't have to pass it as arg to whatever function that wants to return errors. It's inside an option because when constructing the lexer it's empty. I guess the unwrap will be optimized out anyways.

Reply:
Please do it in another commit (could be merged before this one).
Explain clearly the rational and give a proof the the unwrap will be optimized.

Reply:
See #9